addons/pos_coupon/static/src/js/coupon.js

```diff
@@ -274,7 +274,14 @@ odoo.define('pos_coupon.pos', function (require) {
         },
         init_from_JSON: function (json) {
             _order_super.init_from_JSON.apply(this, arguments);
-            this.bookedCouponCodes = json.bookedCouponCodes;
+            this.bookedCouponCodes = {};
+            const filteredLines = json.lines.filter(line => line[2].coupon_id);
```

Comment:
Since you are just parsing the line json it would make sense to move this in the `init_from_JSON` of the orderline
https://github.com/odoo/odoo/blob/d4f1be276d31215158b164f2f282afa8fa75f2b9/addons/pos_coupon/static/src/js/coupon.js#L1046